packages/client/src/generation/generateClient.ts

```diff
@@ -439,3 +444,16 @@ async function getGenerationDirs({ testMode, runtimeDirs, generator, outputDir }
     projectRoot,
   }
 }
+
+/**
+ * Attempts to delete the output directory.
+ * @param finalOutputDir
+ */
+async function deleteOutputDir(finalOutputDir: string) {
+  try {
+    // we want to make sure that if we delete, we delete the right directory
+    if (require(`${finalOutputDir}/package.json`).name === '.prisma/client') {
+      await fs.promises.rmdir(finalOutputDir, { recursive: true })
+    }
+  } catch {}
```

Comment:
We want to make sure that we don't erase wrong things by accident. For this we look that we can find `.prisma/client` in a `package.json` in the output dir, and we only delete the files that we know about/that we generated (TODO).